opentreemap/treemap/udf.py

```diff
@@ -358,11 +358,28 @@ def _update_choice_scalar(self, old_choice_value, new_choice_value):
         self.save()
 
         Model = safe_get_udf_model_class(self.model_type)
-        for model in Model.objects\
-                          .filter(**{'udf:%s' % self.name:
-                                     old_choice_value}):
-            model.udfs[self.name] = new_choice_value
-            model.save_base()
+
+        if self.datatype_dict['type'] == 'choice':
+            for model in Model.objects\
+                              .filter(**{'udf:%s' % self.name:
+                                         old_choice_value}):
+                model.udfs[self.name] = new_choice_value
+                model.save_base()
+
+        else:
+            udf_filter = {'instance': self.instance,
+                          # grab anything with that key
+                          'udf:%s__contains' % self.name: ''}
```

Comment:
There is a hstore-specific way of filtering by key, which is more efficient and more accurate (http://djangonauts.github.io/django-hstore/#_python_api).

Basically, you just wrap the RHS in a list and it will search the hstore column for rows containing those keys:
```
 udf_filter = {'instance': self.instance,
               # grab anything with that key
               'udfs__contains': [self.name]}